### ISSUE TITLE
Warning mle fitting and indication of weights

### ISSUE THREAD

**Issue by lenastroeer** [virocon-organization/virocon#144]
**I'm submitting a:**
  - code improvement request

When using the mle fitting it does not make sense to indicate weights e.g "quadratic".  Accidentally I used the mle fitting together with weights, which did not lead to an error but apparently the weights are just ignored. However, we might need a warning for the user, that indicating weights is not possible with the mle fitting. Weights can only be indicated when using the least squares fitting.

<img width="1232" alt="Bildschirmfoto 2021-11-22 um 13 58 46" src="https://user-images.githubusercontent.com/57865209/142868931-0b01587a-37c5-4535-a278-88e3b2b3dbff.png">

**Comment by kawindme** [comment]
The docstring of the fit method clearly states that weights is ignored for methods other than least squares. 
Would raising a warning be really that helpful?
```python
"""
...
weights : None, str, array_like,
            The weights to use for weighted least squares fitting. Ignored otherwise.
...
"""
```

**Comment by ahaselsteiner** [comment]
Yes, I think a warning would improve the user experience. I am also guilty of not always reading docstrings, however, I read warnings :-)
